src/components/views/admin/BuilderView.tsx

```diff
@@ -39,52 +41,121 @@ import KeyWords from '../../common/settings/KeyWords';
 import SetText from '../../common/settings/SetText';
 import SwitchModes from '../../common/settings/SwitchModes';
 import { useAppSettingContext } from '../../context/AppSettingContext';
+import SyncIcon from './SyncIcon';
+import {
+  DebouncedFunction,
+  SettingKey,
+  SettingValue,
+  defaultSettings,
+  settingKeys,
+} from './types';
+import { getMostRecentTime, hasKeyChanged } from './utils';
 
-const DATA_KEYS = {
-  TEXT: 'text',
-  USE_BOT: 'useBot',
-  KEYWORDS: 'keywords',
-} as const;
-
-const defaultSettings = {
-  [LESSON_TITLE_SETTING_KEY]: {
-    value: DEFAULT_TEXT_RESOURCE_SETTING.text,
-    dataKey: DATA_KEYS.TEXT,
-  },
-  [TEXT_RESOURCE_SETTING_KEY]: {
-    value: DEFAULT_TEXT_RESOURCE_SETTING.text,
-    dataKey: DATA_KEYS.TEXT,
-  },
-  [USE_CHATBOT_SETTING_KEY]: {
-    value: DEFAULT_USE_CHATBOT_SETTING.useBot,
-    dataKey: DATA_KEYS.USE_BOT,
-  },
-  [INITIAL_PROMPT_SETTING_KEY]: {
-    value: DEFAULT_TEXT_RESOURCE_SETTING.text,
-    dataKey: DATA_KEYS.TEXT,
-  },
-  [INITIAL_CHATBOT_PROMPT_SETTING_KEY]: {
-    value: DEFAULT_TEXT_RESOURCE_SETTING.text,
-    dataKey: DATA_KEYS.TEXT,
-  },
-  [KEYWORDS_SETTING_KEY]: {
-    value: [] as Keyword[],
-    dataKey: DATA_KEYS.KEYWORDS,
-  },
-};
-type SettingKey = keyof typeof defaultSettings;
-type SettingValue = (typeof defaultSettings)[SettingKey]['value'];
-
-const settingKeys = Object.keys(defaultSettings).map((k) => k as SettingKey);
+const AUTO_SAVE_DEBOUNCE_MS = 700;
+const REFRESH_SAVE_TIME_MS = 20_000;
```

Comment:
Interesting the `_` what does it do?

Reply:
Nothing, it just allows to improve the visibility for numbers.